### PR TITLE
Bugfix/fix step report

### PR DESCRIPTION
**Pull request by BKaDamien**
always create StepReportData instance even if --step-report flag is not use to ensure the compatibility with test case using step_report variable

### CODE REVIEW

File: src/pykiso/test_coordinator/test_execution.py [hunk 1]
```diff
@@ -199,20 +201,19 @@ def enable_step_report(all_tests_to_run: unittest.suite.TestSuite) -> None:
...
-        # Decorate All assert method
-        assert_method_list = [
-            method for method in dir(tc) if method.startswith("assert")
-        ]
-        for method_name in assert_method_list:
-            # Get method from name
-            method = getattr(tc, method_name)
-            # Add decorator to the existing method
-            setattr(tc, method_name, assert_decorator(method))
+        tc.step_report = StepReportData()
```
Comment (sebastianpfischer) [inline src/pykiso/test_coordinator/test_execution.py:204]: I forgot we got that here... I do not like it

**Review by sebastianpfischer — APPROVED** [review]

File: src/pykiso/test_coordinator/test_execution.py (hunk 1 above)
Comment (sebclrsn) [reply src/pykiso/test_coordinator/test_execution.py:204]: Only other possibility would be to add it by default in BasicTest, BasicTestSuiteSetup and BasicTestSuiteTearDown

**Review by sebclrsn — APPROVED** [review]

File: src/pykiso/test_coordinator/test_execution.py (hunk 1 above)
Comment (BKaDamien) [reply src/pykiso/test_coordinator/test_execution.py:204]: if we do that we have to redesign a little bit our framework

Comment (BKaDamien) [reply src/pykiso/test_coordinator/test_execution.py:204]: sorry it makes two words

Comment (sebastianpfischer) [reply src/pykiso/test_coordinator/test_execution.py:204]: Yea... I was thinking really to cluster it into the test_result section in the future.
and the part where we decorate the `asserts`, we make it a little bit nicer and independent of the step reporter.

**Review by Pog3k — APPROVED** [review]

**Pull request merged**